lib/gcloud/dns/zone/transaction.rb

```diff
@@ -0,0 +1,142 @@
+        #     tx.add     "example.com.", "A",  86400, "1.2.3.4"
+        #     tx.remove  "example.com.", "TXT"
+        #     tx.replace "example.com.", "MX", 86400, ["10 mail1.example.com.",
+        #                                              "20 mail2.example.com."]
+        #   end
+        #
+        def add name, type, ttl, data
+          @additions += Array(@zone.record(name, type, ttl, data))
+        end
+
+        ##
+        # Removes records from the Zone. The records are looked up before they
+        # are removed.
+        #
+        # === Parameters
+        #
+        # +name+::
+        #   The owner of the record. For example: +example.com.+. (+String+)
+        # +type+::
+        #   The identifier of a {supported record
+        #   type}[https://cloud.google.com/dns/what-is-cloud-dns].
+        #   For example: +A+, +AAAA+, +CNAME+, +MX+, or +TXT+. (+String+)
+        #
+        # === Example
+        #
+        #   require "gcloud"
+        #
+        #   gcloud = Gcloud.new
+        #   dns = gcloud.dns
+        #   zone = dns.zone "example-zone"
+        #   zone.update do |tx|
+        #     tx.add     "example.com.", "A",  86400, "1.2.3.4"
+        #     tx.remove  "example.com.", "TXT"
+        #     tx.replace "example.com.", "MX", 86400, ["10 mail1.example.com.",
+        #                                              "20 mail2.example.com."]
+        #   end
+        #
+        def remove name, type
+          @deletions += @zone.records(name: name, type: type).all
+        end
+
+        ##
+        # Replaces existing records on the Zone. Records matching the +name+ and
+        # +type+ are replaced.
+        #
+        # === Parameters
+        #
+        # +name+::
+        #   The owner of the record. For example: +example.com.+. (+String+)
+        # +type+::
+        #   The identifier of a {supported record
+        #   type}[https://cloud.google.com/dns/what-is-cloud-dns].
+        #   For example: +A+, +AAAA+, +CNAME+, +MX+, or +TXT+. (+String+)
+        # +ttl+::
+        #   The number of seconds that the record can be cached by resolvers.
+        #   (+Integer+)
+        # +data+::
+        #   The resource record data, as determined by +type+ and defined in RFC
+        #   1035 (section 5) and RFC 1034 (section 3.6.1). For example:
+        #   +192.0.2.1+ or +example.com.+. (+String+ or +Array+ of +String+)
+        #
+        # === Example
+        #
+        #   require "gcloud"
+        #
+        #   gcloud = Gcloud.new
+        #   dns = gcloud.dns
+        #   zone = dns.zone "example-zone"
+        #   zone.update do |tx|
+        #     tx.add     "example.com.", "A",  86400, "1.2.3.4"
+        #     tx.remove  "example.com.", "TXT"
+        #     tx.replace "example.com.", "MX", 86400, ["10 mail1.example.com.",
+        #                                              "20 mail2.example.com."]
+        #   end
+        #
+        def replace name, type, ttl, data
```

Comment:
I think its a very good idea, just a slightly different use case than replace. With `replace` we don't have to care what was there, just specify what it needs to be. But if you just want to update TTL it would be simpler to use `modify`.

Do you want me to open the issue?